packages/shared/src/types/clerk/types.ts

```diff
@@ -19,3 +19,17 @@ export type UserPublicMetadata = {
   servicesHashes?: IServicesHashes;
   jobTitle?: JobTitleEnum;
 };
+
+/**
+ * Unsafe metadata can be updated from the frontend directly
+ */
+export type UserUnsafeMetadata = {
+  newDashboardOptInStatus?: NewDashboardOptInStatusEnum;
+};
+
+export enum NewDashboardOptInStatusEnum {
+  OPTED_IN = 'opted_in', // user switched to the new dashboard
+  DISMISSED = 'dismissed', // user dismissed the opt-in widget
+  OPTED_OUT = 'opted_out', // user switched back to the old dashboard
```

Comment:
❓ **question:** aren't these two the same? like if I dismissed, I actually opted out

Reply:
How I thought about this is that `opted_out` is that you specifically decided to not use new dashboard after you switched to it and actually seen/tried it.

`dismissed` is when you click the `x` icon on the opt-in widget, that means you didnt visit the new dashboard, you just don't want to see the widget.

I think these two cases have different meaning in regards to tracking and user feedback. Although I would personally not give the option to dismiss the widget, but rather make it smaller and still visible.